crates/spirv-attrib/src/lib.rs

```diff
@@ -1,16 +1,40 @@
-#![cfg_attr(target_arch = "spirv", feature(register_attr), register_attr(spirv))]
+use core::iter::FromIterator;
+use proc_macro::{Delimiter, Group, TokenStream, TokenTree};
 
-use proc_macro::TokenStream;
-use std::str::FromStr;
-
-#[cfg(not(target_arch = "spirv"))]
 #[proc_macro_attribute]
 pub fn spirv(_attr: TokenStream, item: TokenStream) -> TokenStream {
-    let mut item_str = item.to_string();
+    let mut tokens = Vec::new();
+    for tt in item {
+        match tt {
+            TokenTree::Group(group) => match group.delimiter() {
+                Delimiter::Parenthesis => {
+                    let mut sub_tokens = Vec::new();
+                    for tt in group.stream() {
+                        match tt {
+                            TokenTree::Group(group) => match group.delimiter() {
+                                Delimiter::Bracket => {
+                                    if group.stream().to_string().starts_with("spirv") {
+                                        sub_tokens.pop();
+                                    } else {
+                                        sub_tokens.push(TokenTree::from(group));
+                                    }
+                                }
+                                _ => sub_tokens.push(TokenTree::from(group)),
+                            },
+                            _ => sub_tokens.push(tt),
```

Comment:
You only need the last one (and outside the `match`) if you make the one that skips use `continue;`.
To not consume `tt` you can use `match &tt` instead of `match tt`.

Reply:
done